### PR TITLE
Minor improvements to C# code quality.

### PR DESCRIPTION
**Pull request by cgranade**

### CODE REVIEW

**Review by anjbur — COMMENTED** [review]
In PauliTerm.cs, the call to `ReferenceEquals` on line 178 (formerly line 199) was also flagged. Otherwise looks good to me!

**Comment by cgranade** [comment]
Thanks for the review, @anjbur! I think that should now be fixed. There was also a build failure caused by not implementing `IEquatable<T>` correctly, so I went on and fixed that as well by using a pattern match on `IEquatable<T>` from `object.Equals`. Thanks again!

File: Standard/src/Emulation/Math.cs [hunk 1]
```diff
@@ -192,9 +194,9 @@ private long SampleUsingBTPE()
             if (v > F) goto Step10;
             goto Step60;
 
-            Step52:
+        Step52:
             rho = (k/(nrq))*((k*(k/3.0 + 0.625) + 0.16666666666666666)/nrq + 0.5);
-            t = -k*k/(2*nrq);
+            t = -k*k/(2*nrq); // lgtm [cs/loss-of-precision]
```
Comment (guenp) [inline Standard/src/Emulation/Math.cs:199]: Not sure I understand this comment?

Comment (cgranade) [reply Standard/src/Emulation/Math.cs:199]: It's an instruction to CI to indicate that a warning raised about that line is a false positive.

**Review by msoeken — APPROVED** [review]

**Review by anjbur — APPROVED** [review]
Looks good to me, thank you!

**Review by anpaz — APPROVED** [review]
lgtm.

**Pull request merged**